grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannel.java

```diff
@@ -356,14 +372,15 @@ static String getKeyFromMessage(MessageOrBuilder msg, String name) {
     if (affinity != null) {
       AffinityConfig.Command cmd = affinity.getCommand();
       String keyName = affinity.getAffinityKey();
-      String key = getKeyFromMessage((MessageOrBuilder) message, keyName);
-      if (key != null && key != "") {
-        if ((isReq && (cmd == AffinityConfig.Command.UNBIND || cmd == AffinityConfig.Command.BOUND))
-            || (!isReq && cmd == AffinityConfig.Command.BIND)) {
-          {
-            return key;
-          }
+      List<String> keys = getKeyFromMessage((MessageOrBuilder) message, keyName);
+      if (isReq && (cmd == AffinityConfig.Command.UNBIND || cmd == AffinityConfig.Command.BOUND)) {
+        if (keys.size() > 1) {
```

Comment:
`keys.size()` can be 0 as well, as I can see from the logic in `getKeyFromMessage`.

Also, can we also support multiple unbindings? Spanner may have BatchDeleteSessions as well in future.

Reply:
Done